scaling_performance/optimizing_storage.adoc

```diff
@@ -12,6 +12,86 @@
+|=======================
+NOTE: As of {product-title} 3.6.1, CNS GlusterFS (Container-Native Storage, a hyperconverged or cluster-hosted storage solution) and CRS GlusterFS (Container-Ready Storage, an externally hosted storage solution) provides interfaces for Block Storage, File Storage and Object Storage for the purpose of the {product-title} Registry, Logging, and Metrics.
+
+NOTE: ^(1)^ CNS/CRS GlusterFS, Ceph RBD, OpenStack Cinder, AWS EBS, Azure Disk, GCE Persistent Disk, VMWare vSphere support Dynamic PV Provisioning natively in {product-title}.
+
+NOTE: ^(2)^ NetApp NFS supports Dynamic PV Provisioning when using the Trident Plugin
+
+WARNING: ^(3)^ Vendor GlusterFS, Vendor S3, and Vendor Swift supportability and configurability may vary.
+
+=== Storage Backend Recommendations
+Below is a matrix that summarizes which storage backend types are Recommended and Configurable for the given
+{product-title} cluster application and useful attributes.
+[format="csv",width="70%",cols="8"]
+[options="header]
+|=======================
+Storage Type,ROX,RWX,Registry,Scaled Registry,Metrics,Logging,Apps
+Block,Yes^(1)^,No,Configurable,Not Configurable,Recommended,Recommended,Recommended
+File,Yes^(1)^,Yes,Configurable,Configurable,Configurable,Configurable,Recommended
+Object,Yes,Yes,Recommended,Recommended,Not Configurable,Not Configurable,Not Configurable^(2)^
+|=======================
+NOTE: A Scaled Registry is a {product-title} Registry where 3 or more pod replicas are running.
+
+NOTE: ^(1)^ This does not apply to Physical Disk, VM Physical Disk, VMDK, Loopback over NFS, AWS EBS, and Azure Disk.
+
+NOTE: ^(2)^ Object Storage is not consumed via {product-title}'s PVs/PVCs.  Apps must integrate with the Object Storage REST API.
+
+==== Specific Application & Storage Recommendations
+===== Registry
+In a non-scaled/HA {product-title} Registry cluster deployment, the preferred Storage Backend is Object Storage
+followed by Block Storage.  The Storage Backend does not need to support ReadWriteMany Access Mode (RWX).  The Storage Backend *must
+ensure Read-After-Write-Consistency*.  All Network Attached File Storage Backends (excluding CNS/CRS GlusterFS) are expressly *not recommended* for {product-title} Registry cluster deployment with production workloads.  While hostPath volumes are configurable for a non-scaled/HA {product-title} Registry cluster deployment, they are not recommended.
+
+WARNING: Corruption may occur when using NFS to back {production-title} Registry at production workloads.
+
+===== Scaled Registry
+In a scaled/HA {product-title} Registry cluster deployment, the preferred Storage Backend is Ojbect Storage.
+The Storage Backend *must support ReadWriteMany* Access Mode (RWX) and *must ensure Read-After-Write-Consistency*.  Block Storage is
```

Comment:
> Block Storage is expressly *not recommended* for a scaled/HA {product-title} Registry cluster deployment with production workloads

I think here you mean file File instead of Block. I don't think we have a single block storage that allows RWX.

Reply:
Whoops.  You're right.  Too much copy-paste on my part.  And actually, it's both File & Block that are not recommended.